src/Drawer/Drawer.tsx

```diff
@@ -35,12 +35,19 @@ export const Drawer = React.forwardRef<HTMLDivElement, DrawerProps>(
     }, [state.drawerStep]) // eslint-disable-line react-hooks/exhaustive-deps
     React.useEffect(() => {
       if (stateProp && stateProp !== state.drawerStep) {
-        dispatch({
-          type: DrawerActionTypes.ChangeStep,
-          value: stateProp,
-          containerHeight: mergedRef.current?.clientHeight ?? 0,
-        })
+        const initializedStep = () => {
```

Comment:
You should debounce this function since it's being called on every `resize` events.